pkg/tarball/tarball.go

```diff
@@ -25,8 +25,35 @@ import (
 	"time"
 )
 
+type Context struct {
+	SourceDateEpoch time.Time
+}
+
+type Option func(*Context) error
+
+// Generates a new context from a set of options.
```

Comment:
I'm not going to approve because I'm too much of a newb to both Go and this project, but I thought I should point out (as a totally optional point) that the Godoc tool and Go documentation convention for a function is to start sentences with the function's name: `NewContext generates a new context from a set of options.` Thought this might help with documentation later. If this isn't helpful, just ignore. [Relevant official Go blog post on Godoc and documentation](https://go.dev/blog/godoc).